packages/eslint-plugin/src/rules/prefer-optional-chain.ts

```diff
@@ -48,6 +48,59 @@ export default util.createRule({
   create(context) {
     const sourceCode = context.getSourceCode();
     return {
+      'LogicalExpression[operator=||],LogicalExpression[operator=??]'(
+        node: TSESTree.LogicalExpression,
+      ): void {
+        const leftNode = node.left;
+        const rightNode = node.right;
+        const parentNode = node.parent;
```

Comment:
Just nitpicking: I can see why you made the two bigger variables below to make the code more clear, but at least for these three the single character of name savings / object member lookup isn't worth the extra code bloat.

Not a blocker IMO (someone should eventually write a lint rule if it's important), just a thought. :)

Reply:
It started out with `parentNode` for reuse and typescript's type narrowing inside the fixer (typescript doesn't narrow property's type inside other functions' scopes if the type guard is above the function).

The left & right nodes are just a stylistic choice to match it.